### ISSUE TITLE
create a "big" partition

### ISSUE THREAD

**Issue by ericleasemorgan** [ericleasemorgan/reader#68]
Our system current has a partition called "cloud". We need and can afford to use more cores than what the cloud partition provides. On Distant Reader Classic there is a partition called "big-cloud". Please create such a partition on dr-cov19-headnode. 'Make sense? @ECoulter

**Comment by ECoulter** [comment]
This actually already existed on the DR-COV-19 headnode! I should've realized, since it's a direct clone of the DR-Classic. Job submitted to the queue 'big-compute' will run on a single worker node with 24 cores.

**Comment by ericleasemorgan** [comment]
This is great, but alas, only one of these big nodes seems to be able to run at a single time:

             JOBID PARTITION     NAME     USER ST	TIME  NODES NODELIST(REASON)
              2457 big-cloud treatmen  emorgan PD	0:00	  1 (Resources)
              2456 big-cloud united-s  emorgan  R	7:48	  1 big-compute-0

Is there any way things can be configured so multiple big-cloud nodes can be run simultaneously? Does our project have those kind of resources?

--
Eric M.

**Comment by ECoulter** [comment]
We can support one of
4 "big", 3 "regular" 
3 "big", 5 "regular"
2 "big", 7 "regular"
1 "big,  9 "regular" 
nodes - the big ones are 24 cpus, small are 10 cpus, the max (due to the Jetstream quota) is 132 cpus, and we have currently 10 cpus used by other infrastructure. 

AFAIK, Slurm isn't clever enough to handle allocating only a max number of CPUs across partitions, so it's easiest to keep the total possible number of cores below our JS quota. 

Not sure which distribution you'd like to go with, but modulo information on the expected types of jobs, I'll vote for the 3/5 split as a matter of aiming for the middle.

**Comment by ericleasemorgan** [comment]
On Jun 3, 2020, at 1:13 PM, Eric Coulter <notifications@github.com> wrote:

> We can support one of:
> 
>   4 "big", 3 "regular"
>   3 "big", 5 "regular"
>   2 "big", 7 "regular"
>   1 "big, 9 "regular"
> 
> nodes - the big ones are 24 cpus, small are 10 cpus, the max (due to the Jetstream quota) is 132 cpus, and we have currently 10 cpus used by other infrastructure.
> 
> AFAIK, Slurm isn't clever enough to handle allocating only a max number of CPUs across partitions, so it's easiest to keep the total possible number of cores below our JS quota.
> 
> Not sure which distribution you'd like to go with, but modulo information on the expected types of jobs, I'll vote for the 3/5 split as a matter of aiming for the middle.


I like your vote regarding the distribution -- 3/5. I suppose you need to do some configurations in order for that to happen. Correct? If so, then I will wait to hear from you. When we get that far, I will edit a script to take advantage of the distribution. Okay? --Eric M.

**Comment by ECoulter** [comment]
It is done! That's a fairly easy configuration change. Let me know if you have any troubles!

**Comment by ericleasemorgan** [comment]
Hmmm... When I submit a job a big node appears, but when I submit a second job the processing happens on the same big node. I expected a new node to be created big-compute-0, big-compute-1, etc. Are my expectations incorrect? --Eric M.

**Comment by ECoulter** [comment]
Ah, so the job I see on there now only request 1 CPU - so Slurm thinks there's room for 23 more jobs. 
Two immediate solutions:
Add -n 24 to the sbatch script
or
set big-compute so that only 1 job is allowed per node.

**Comment by ECoulter** [comment]
(your expectations were not incorrect!)

**Comment by ericleasemorgan** [comment]
> On Jun 3, 2020, at 5:17 PM, Eric Coulter <notifications@github.com> wrote:
> 
> 
> Ah, so the job I see on there now only request 1 CPU - so Slurm thinks there's room for 23 more jobs.
> Two immediate solutions:
> Add -n 24 to the sbatch script
> or
> set big-compute so that only 1 job is allowed per node.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 


I'm sorry, but I still don't quite get it. I have configuration like this:

  # sbatch configurations
  #SBATCH -p big-cloud
  #SBATCH -N 1
  #SBATCH -n 20
  #SBATCH -t 48:0:00
  #SBATCH -J united-states
  #SBATCH -o standard-output.txt
  #SBATCH -e standard-error.txt
  #SBATCH --mail-user=emorgan@nd.edu
  #SBATCH --mail-type=ALL

Now three nodes will spin up but two of them are in the PD state.

--
Eric M.

**Comment by ericleasemorgan** [comment]
P.S. The cloud partition works as expected:

   JOBID PARTITION     NAME     USER ST	TIME  NODES NODELIST(REASON)
    2468 big-cloud    italy  emorgan PD	0:00	  1 (Resources)
    2469 big-cloud    spain  emorgan PD	0:00	  1 (Priority)
    2467 big-cloud united-s  emorgan  R	8:58	  1 big-compute-0
    2470     cloud lock-dow  emorgan  R	3:04	  1 compute-0
    2471     cloud    china  emorgan  R	3:04	  1 compute-1

I think something needs to be reset? 'Still learning, I am, about Slurm.

--
Eric Backwards Talk

**Comment by ECoulter** [comment]
Ah, that was my fault - I didn't realize the slurm_resume script on there was lacking an update, which allows me to update the slurm.conf without updating the compute node image - I'm working on applying it.

**Comment by ericleasemorgan** [comment]
I hate to re-open this issue, but the nodes seem to be in an almost constant state of CF:

  JOBID PARTITION     NAME     USER ST	TIME  NODES NODELIST(REASON)
   2562 big-cloud test-ppe  emorgan CF      10:09	  1 big-compute-1
   2563 big-cloud test-soc  emorgan CF      10:02	  1 big-compute-2
   2558 big-cloud   people  emorgan  R    5:35:30	  1 big-compute-0

Is this to be expected? 

--
ELM

**Comment by ECoulter** [comment]
Nope, that was not expected - still had an error in the slurm_resume script - a missing linebreak, of all things (copy-paste error on my part, and insufficient testing). I fixed this, and tested in big-cloud on multiple nodes, so I *think* the bugs are finally squashed.

**Comment by ericleasemorgan** [comment]
I did that one time. I put a carriage return in the wrong place and consequently locked myself out of my own machine. The host was co-located four hours away. There was no way I could ssh to my own machine. Zero. My computer was locked down so tight that it was rusted shut. I had to drive to Detroit, gain access to the machine room, have a guy look over my shoulder making sure I wasn't doing anything wrong, and add a carriage return to /etc/hosts.allow  It took me longer to walk from the parking lot to my computer than it did for me to fix things.

In any event, issue resolved, we now have three big-cloud partitions running concurrently. Thank you.
